internal/plugincommon/discovery/processor.go

```diff
@@ -182,28 +180,65 @@ func aggregateObservations(
 	return obs
 }
 
-// Outcome comes to consensus on the contract addresses and updates the chainreader. It doesn't actually
-// return an Outcome.
+// getRouterConsensus figures out a single router address from the multiple observed addresses from multiple source
+// chains. This happens in two phases, first we get the consensus on the router address for each source chain
+// according the threshold (fChain). Second, once we have consensus across the source chains, we select whichever
+// address was observed the most times. If there is a tie, we return an error.
+//
+// For a typical configuration, we expect all source chains to report the same destination router address.
```

Comment:
I think this comment is confusingly worded because we are _not_ reading source chain routers - we are reading the dest chain's local router reference for messages coming from each source chain (see the link I pasted below).

I've attempted to clarify this below but let me know if there's any further misunderstandings.

```suggestion
// getRouterConsensus figures out a single dest router address from the multiple observed addresses from multiple offramp source
// chain configurations. This happens in two phases, first we get the consensus on the router address for 
// each offramp source chain config according the dest 2*fChain + 1 threshold. Second, once we have consensus across the source chains, 
// we select whichever address was observed the most times. If there is a tie, we return an error.
//
// For a typical configuration, we expect all offramp source chain configurations to report the same router address.
// See https://github.com/smartcontractkit/chainlink/blob/05c8b45da51340aa9575ac909486decc9c164792/contracts/src/v0.8/ccip/offRamp/OffRamp.sol#L105
```